scripts/exwafs_grib2_0p25_blending.sh

```diff
@@ -2,152 +2,100 @@
-    ukfiles=$(find "${COMINuk}" -name "egrr_wafshzds_unblended_*_0p25_${PDY:0:4}-${PDY:4:2}-${PDY:6:2}T${cyc}:00Z_t${fhr}.grib2" | wc -l)
-    if ((ukfiles >= 3)); then
-        echo "Found all 3 UK WAFS GRIB2 files, continue ..."
-        break
-    fi
+fhours=($(seq -s ' ' -f "%03g" 6 1 24; seq -s ' ' -f "%03g" 27 3 48))
+np=${#fhours[@]}
+MPIRUN="mpiexec -np ${np} -cpu-bind verbose,core cfp"
 
-    if ((ic == SLEEP_LOOP_MAX)); then
-        products="cb ice turb"
-        for prod in ${products}; do
-            ukfile="egrr_wafshzds_unblended_${prod}_0p25_${PDY:0:4}-${PDY:4:2}-${PDY:6:2}T${cyc}:00Z_t${fhr}.grib2"
-            if [[ ! -f "${COMINuk}/${ukfile}" ]]; then
-                echo "WARNING: UK WAFS GRIB2 file '${ukfile}' not found after waiting over ${SLEEP_TIME} seconds"
-                echo "${COMINuk}/${ukfile}" >>missing_uk_files
-            fi
-        done
-        echo "WARNING: UK WAFS GRIB2 unblended data is not completely available, no blending"
-        SEND_UNBLENDED_US_WAFS="YES"
-        break
-    else
-        sleep "${SLEEP_INT}"
-    fi
+rm -f wafsgrib2_0p25.cmdfile
+for fhr in ${fhours[@]}; do
+    echo "${USHwafs}/wafs_grib2_0p25_blending.sh $fhr > $DATA/${fhr}.log 2>&1">> wafsgrib2_0p25_blending.cmdfile
+    export MP_PGMMODEL=mpmd
 done
+$MPIRUN wafsgrib2_0p25_blending.cmdfile
 
-##########################
-# Blending or unblended
-##########################
-
-if [[ "${SEND_UNBLENDED_US_WAFS}" == "YES" ]]; then
-    echo "turning back on dbn alert for unblended US WAFS product"
-else
-    # retrieve UK products
-    # Three(3) unblended UK files for each cycle+fhour: icing, turb, cb
-    cat "${COMINuk}/egrr_wafshzds_unblended_"*"_0p25_${PDY:0:4}-${PDY:4:2}-${PDY:6:2}T${cyc}:00Z_t${fhr}.grib2" >"EGRR_WAFS_0p25_unblended_${PDY}_${cyc}z_t${fhr}.grib2"
-
-    # pick up US data
-    cpreq "${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2" .
+export err=$? ; err_chk
 
-    # run blending code
-    export pgm="wafs_blending_0p25.x"
+for fhr in ${fhours[@]}; do
+    echo "=================== log file of fhr=$fhr ==================="
+    cat "${DATA}/${fhr}.log"
+done
+echo "===================== end of log files ====================="
 
-    . prep_step
+missing_uk_files="$(find $DATA -name 'missing_uk_files*')"
+missing_us_files="$(find $DATA -name 'missing_us_file*')"
+no_blending_files="$(find $DATA -name 'no_blending_files*')"
 
-    ${EXECwafs}/${pgm} "WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2" \
-        "EGRR_WAFS_0p25_unblended_${PDY}_${cyc}z_t${fhr}.grib2" \
-        "0p25_blended_${PDY}${cyc}f${fhr}.grib2 >f${fhr}.out"
-    err=$?
-    if ((err != 0)); then
-        echo "WARNING: WAFS blending 0p25 program failed at '${PDY}${cyc}f${fhr}'. Turning back on dbn alert for unblended US WAFS product"
-        SEND_UNBLENDED_US_WAFS="YES"
+if [[ ! -z "$missing_uk_files" ]] || [[ ! -z "$missing_us_files" ]] || [[ ! -z "$no_blending_files" ]] ; then
```

Comment:
I don't think this logic is correct.
When there are no US and no UK, missing_uk_files and missing_us_files will be empty, and only no_blending files will be present.

On the other hand, if only UK is absent, this block will still send the same alert to AWC, even though US is present

Reply:
If no_blending files, dbn_alert and email will be sent, causing different attentions of NCO. 
For example, at f007 if both UK and US are missing:
 1. At all other fhours, UK and US are present. Then only no_blending is alerted and emailed.
 2. UK data is absent at other fhours, both ukmissing and no_blending is alerted and emailed.